holoviews/plotting/bokeh/element.py

```diff
@@ -177,15 +177,18 @@ def _construct_callbacks(self):
         the plotted object as a source.
         """
         if not self.static or isinstance(self.hmap, DynamicMap):
-            source = self.hmap
+            sources = [(i, o) for i, o in get_sources(self.hmap)
+                       if i in [None, self.zorder]]
         else:
-            source = self.hmap.last
-        streams = Stream.registry.get(id(source), [])
-        registry = Stream._callbacks['bokeh']
-        callbacks = {(registry[type(stream)], stream) for stream in streams
-                     if type(stream) in registry and streams}
+            sources = [(self.zorder, self.hmap.last)]
+        cb_classes = set()
+        for _, source in sources:
```

Comment:
So the key difference seems to be that you can now have multiple sources whereas before there was only one. And the sources are now retrieved recursively via ``get_sources`` which works via the ``Callable`` instances...